packages/internet-header/src/components/post-internet-header/post-internet-header.tsx

```diff
@@ -174,48 +201,127 @@ export class PostInternetHeader {
   }
 
   async componentWillLoad() {
+    // Validate project ID
+    if (this.project === undefined || this.project === '' || !isValidProjectId(this.project)) {
+      throw new Error(
+        `Internet Header project key is invalid. Please provide a valid project key.`
+      );
+    }
+
+    if (!this.isValidEnvironment(this.environment)) {
+      throw new Error(
+        `Internet Header environment "${this.environment}" is not valid. Please use one of: ${PostInternetHeader.VALID_ENVIRONMENTS.join(', ')}`
+      );
+    }
```

Comment:
If you give a default value to the parameter of the `handleEnvironmentChange` function like so: 
```ts
  handleEnvironmentChange(newValue = this.environment) {
```

You can completely reuse it in the `componentWillLoad`, intead of duplicating the error message: 
```suggestion
    this.handleEnvironmentChange();
```

It's usually how we do it in our components, here the example of the `post-banner`: 
https://github.com/swisspost/design-system/blob/main/packages/components/src/components/post-banner/post-banner.tsx#L87